packages/api/tests/lambdas/test-process-s3-dead-letter-archive.js

```diff
@@ -152,9 +152,76 @@ test('processDeadLetterArchive only deletes dead letters that process successful
     writeRecordsFunction: writeRecordsErrorThrower,
   });
 
-  const remainingDeadLetters = await S3.listS3ObjectsV2({ Bucket: bucket, Prefix: path });
-  t.is(remainingDeadLetters.length, 1);
+  // Check that processed message key was deleted
+  const processedDeadLetterExists = await S3.fileExists(bucket, processedMessageKey);
+  t.is(processedDeadLetterExists, false);
+  t.deepEqual(output.processingSucceededKeys, [processedMessageKey]);
+});
+
+test('processDeadLetterArchive saves failed dead letters to different S3 and removes from previous S3 path', async (t) => {
+  const {
+    bucket,
+    path,
+    messageKeys,
+  } = t.context;
+  const passingMessageExecutionName = getMessageExecutionName(t.context.cumulusMessages[1]);
+  const failingMessageKey = messageKeys[0];
+  const s3KeyForFailedMessage = generateNewArchiveKeyForFailedMessage(failingMessageKey);
+  const writeRecordsErrorThrower = ({ cumulusMessage }) => {
+    if (getMessageExecutionName(cumulusMessage) === passingMessageExecutionName) return;
+    throw new Error('write failure');
+  };
+
+  const output = await processDeadLetterArchive({
+    bucket,
+    path,
+    writeRecordsFunction: writeRecordsErrorThrower,
+  });
+
+  // Check that failing message key was deleted
+  const remainingDeadLetterExists = await S3.fileExists(bucket, failingMessageKey);
```

Comment:
can you rename `remainingDeadLetterExists` to `failingMessageRemainInOldLocation` or something more understandable?

Reply:
https://github.com/nasa/cumulus/pull/3099/commits/a738f3f1cf4a78d1454c73c6654d2d47938d341a#diff-ef5dda899c4f3f23dcc7486c3e5143a997544e8348bfa2241a444f3921fdad3aR182